src/index.js

```diff
@@ -3,10 +3,10 @@
 var multiaddr = require('multiaddr')
 
 var IP = or(base('ip4'), base('ip6'))
-var TCP = and(IP, base('tcp'))
-var UDP = and(IP, base('udp'))
-var UTP = and(UDP, base('utp'))
-var WebSockets = and(TCP, base('websockets'))
+var TCP = and(IP, base('tcp', 'ipfs'))
+var UDP = and(IP, base('udp', 'ipfs'))
+var UTP = and(UDP, base('utp', 'ipfs'))
+var WebSockets = and(TCP, base('websockets', 'ipfs'))
```

Comment:
Because that's how I implemented it, so that `base` takes an optional second thing